### PR TITLE
Defer Creation of XmlMapper to Allow for Exclusion of Dataformat XML

### PR DESCRIPTION
**Pull request by alzimmermsft**
# Description

Refactors `JacksonAdapter`/`ObjectMapperFactory` so that all code referencing Jackson Dataformat XML is in a standalone file allowing for projects using `azure-core`, or other Azure SDKs built on top of `azure-core`, to exclude `com.fasterxml.jackson.dataformat:jackson-dataformat-xml` from the dependency list. NOTE, if the project ends up using XML serialization in `JacksonAdapter` and Jackson Dataformat XML is excluded that application will crash with `ClassNotFoundException`s and `MethodNotFoundException`s, this change is purely to support scenarios where an application is known to never require XML serialization.

Example POM using this change:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
  <modelVersion>4.0.0</modelVersion>

  <groupId>org.example</groupId>
  <artifactId>ExcludeDataformatXml</artifactId>
  <version>1.0-SNAPSHOT</version>

  <properties>
    <maven.compiler.source>11</maven.compiler.source>
    <maven.compiler.target>11</maven.compiler.target>
    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
  </properties>

  <dependencies>
    <dependency>
      <groupId>com.azure</groupId>
      <artifactId>azure-core</artifactId>
      <version>1.31.0-beta.1</version>
      <exclusions>
        <exclusion>
          <groupId>com.fasterxml.jackson.dataformat</groupId>
          <artifactId>jackson-dataformat-xml</artifactId>
        </exclusion>
      </exclusions>
    </dependency>

    <dependency>
      <groupId>com.azure</groupId>
      <artifactId>azure-search-documents</artifactId>
      <version>11.5.0-beta.13</version>
      <exclusions>
        <exclusion>
          <groupId>com.fasterxml.jackson.dataformat</groupId>
          <artifactId>jackson-dataformat-xml</artifactId>
        </exclusion>
      </exclusions>
    </dependency>
  </dependencies>

</project>
```

In this XML the application is using Azure Cognitive Search which is a JSON-based service and doesn't require the use of Jackson Dataformat XML. The next example is a case where the application would crash due to XML serialization being required:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
  <modelVersion>4.0.0</modelVersion>

  <groupId>org.example</groupId>
  <artifactId>ExcludeDataformatXml</artifactId>
  <version>1.0-SNAPSHOT</version>

  <properties>
    <maven.compiler.source>11</maven.compiler.source>
    <maven.compiler.target>11</maven.compiler.target>
    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
  </properties>

  <dependencies>
    <dependency>
      <groupId>com.azure</groupId>
      <artifactId>azure-core</artifactId>
      <version>1.31.0-beta.1</version>
      <exclusions>
        <exclusion>
          <groupId>com.fasterxml.jackson.dataformat</groupId>
          <artifactId>jackson-dataformat-xml</artifactId>
        </exclusion>
      </exclusions>
    </dependency>

    <dependency>
      <groupId>com.azure</groupId>
      <artifactId>azure-storage-blob</artifactId>
      <version>12.19.0</version>
      <exclusions>
        <exclusion>
          <groupId>com.fasterxml.jackson.dataformat</groupId>
          <artifactId>jackson-dataformat-xml</artifactId>
        </exclusion>
      </exclusions>
    </dependency>
  </dependencies>

</project>
```

Azure Storage Blob is an XML-based service and when the first attempt to make a call to the service is made the application will crash as XML serialization won't be available.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/core/azure-core/src/main/java/com/azure/core/implementation/jackson/XmlMapperFactory.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        } catch (Throwable ex) {
+            // Throw the Error only if it isn't a LinkageError.
+            // This initialization is attempting to use classes that may not exist.
+            if (ex instanceof Error && !(ex instanceof LinkageError)) {
+                throw (Error) ex;
+            }
+
+            LOGGER.verbose("Failed to retrieve MethodHandles used to set coercion configurations. "
+                + "Setting coercion configurations will be skipped. "
+                + "Please update your Jackson dependencies to at least version 2.12", ex);
```
Comment (lmolkova) [inline sdk/core/azure-core/src/main/java/com/azure/core/implementation/jackson/XmlMapperFactory.java:57]: nit:
```suggestion
                + "Please update your Jackson dependencies to at least version", ex);
```

**Review by lmolkova — APPROVED** [review]

File: sdk/core/azure-core/src/main/java/com/azure/core/implementation/jackson/XmlMapperFactory.java (hunk 1 above)
Comment (alzimmermsft) [reply sdk/core/azure-core/src/main/java/com/azure/core/implementation/jackson/XmlMapperFactory.java:57]: I'm not quite sure I understand this nit, is this asking to remove the `2.12` mention? But nothing is added back and the `MethodHandle`s require at least version 2.12 to be created properly.

Comment (lmolkova) [reply sdk/core/azure-core/src/main/java/com/azure/core/implementation/jackson/XmlMapperFactory.java:57]: you're right, never mind

File: sdk/servicebus/azure-messaging-servicebus/pom.xml [hunk 2]
```diff
@@ -113,4 +119,23 @@
...
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-enforcer-plugin</artifactId>
+        <version>3.0.0-M3</version> <!-- {x-version-update;org.apache.maven.plugins:maven-enforcer-plugin;external_dependency} -->
+        <configuration>
+          <rules>
+            <bannedDependencies>
```
Comment (ki1729) [inline sdk/servicebus/azure-messaging-servicebus/pom.xml:131]: Nice one!

**Pull request merged**